### PR TITLE
[front] feature(vault): Data source view in apps

### LINKED ISSUE #7154: [Dust Apps] start using dsViews in Dust apps DATASOURCE block

### PR DESCRIPTION
**Pull request by tdraier**
fixes: https://github.com/dust-tt/dust/issues/7154

## Description

- Change dataSourcePicker to select a dataSource view in the current vault instead of looking in all datasources.
- Set datasource view id in app config
- Edit link to datasource in vault or builder (depending on feature flag)
- Add Vault in AppType

## Deploy Plan

deploy `front`

### CODE REVIEW

**Review by fontanierh — APPROVED** [review]
LGTM 🙏

File: front/components/data_source/DataSourcePicker.tsx [hunk 1]
```diff
@@ -24,62 +30,89 @@ export default function DataSourcePicker({
...
+    vaultId: vault.sId,
+    workspaceId: owner.sId,
+  });
 
   const [searchFilter, setSearchFilter] = useState("");
-  const [filteredDataSources, setFilteredDataSources] = useState(dataSources);
+  const [filteredDataSourceViews, setFilteredDataSourceViews] =
+    useState(vaultDataSourceViews);
+
+  const selectedDataSourceView = hasDataSourceView
```
Comment (spolu) [inline front/components/data_source/DataSourcePicker.tsx:53]: Probably worth a small comment here since this is what does all the conversion magic

**Comment by spolu** [comment]
@tdraier we have 404 when clicking on the data source views's data source name in the picker also this visual bug?

**Comment by spolu** [comment]
Can we fix the search design as well it's really under-par :pray:

**Comment by tdraier** [comment]
> Do we have a plan to migrate the savedConfig also?

I'll create a task for that

**Pull request merged**